eZ/Publish/Core/Base/Exceptions/DatabaseException.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+/**
+ * Contains Database Exception implementation.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Base\Exceptions;
+
+use RuntimeException;
+
+/**
+ * An exception that is thrown when the database encounters an error.
+ */
+final class DatabaseException extends RuntimeException
+{
+    public const MESSAGE = 'Database error';
```

Comment:
```suggestion
    public const DEFAULT_MESSAGE = 'Database error';
```

IMHO to better DX we should override __constructor and use `DEFAULT_MESSAGE` as default `$message` parameter value. Also to simplify usage even more, we might intoduce a factory method:

```php
public static function wrap(Throwable $previous, $message = self::DEFAULT_MESSAGE, $code = 0): self
{
    return new self($message, $code, $previous);
}
```